tree/treeplayer/inc/TFormLeafInfo.h

```diff
@@ -44,6 +44,26 @@
    template<typename T> T  ReadValueImpl(char *where, Int_t instance = 0)  // no semicolon
 
 
+// declare the extra versions of GetValue() plus templated implementation
+#define DECLARE_GETVAL_OVERRIDE \
+   Double_t  GetValue(TLeaf *leaf, Int_t instance = 0) override               \
+       { return GetValueImpl<Double_t>(leaf, instance); }                     \
```

Comment:
What about using only one (well 2) macros:
```suggestion
#define DECLARE_GETVAL(VIRTUAL, OVERRIDE) \
   VIRTUAL Double_t  GetValue(TLeaf *leaf, Int_t instance = 0) OVERRIDE               \
       { return GetValueImpl<Double_t>(leaf, instance); }                                              \
```
and usage:
```
DECLARE_GETVAL(virtual,)
// or
DECLARE_GETVAL( , override)
```

Reply:
I agree, it's a nice way to avoid the code duplication! I have updated the PR.